code/modules/xenoarchaeology/xenoartifact.dm

```diff
@@ -0,0 +1,489 @@
+    check_charge()
+
+/obj/item/xenoartifact/proc/set_frequency(new_frequency)
+	SSradio.remove_object(src, frequency)
+	frequency = new_frequency
+	radio_connection = SSradio.add_object(src, frequency, RADIO_SIGNALER)
+
+/obj/item/xenoartifact/proc/send_signal(var/datum/signal/signal)
+    if(!radio_connection||!signal)
+        return
+    radio_connection.post_signal(src, signal)
+
+/obj/item/xenoartifact/receive_signal(datum/signal/signal)
+    if(!signal || signal.data["code"] != code)
+        return
+    var/mob/living/M = isliving(signal.source.loc) ? signal.source.loc : null
+    audible_message("[icon2html(src, hearers(src))] *beep* *beep* *beep*", null, 3)
+    playsound(get_turf(src), 'sound/machines/triple_beep.ogg', ASSEMBLY_BEEP_VOLUME, TRUE)
+    for(var/datum/xenoartifact_trait/T in traits)
+        if(charge += EASY*T.on_signal(src))
+            true_target += list(get_proximity(max_range))
+            check_charge(M)
+
+/obj/item/xenoartifact/on_block(mob/living/carbon/human/owner, atom/movable/hitby)
+    . = ..()
+    if(!(manage_cooldown(TRUE)))
+        return
+    for(var/datum/xenoartifact_trait/T in traits)
+        if(charge += EASY*T.on_impact(src))
+            true_target += list(process_target(hitby))
+            check_charge(owner)
+    return
+
+/obj/item/xenoartifact/process(delta_time)
+    switch(process_type)
+        if("lit")
+            true_target += list(get_proximity(min(max_range, 5)))
+            charge = NORMAL*traits[1].on_burn(src) 
+            if(manage_cooldown(TRUE) && true_target.len >= 1 && get_proximity(max_range))
+                set_light(0)
+                visible_message("<span class='danger'>The [name] flicks out.</span>")
+                check_charge()
+                process_type = ""
+                return PROCESS_KILL
+        if("tick")
+            true_target += list(get_proximity(max_range))
+            if(manage_cooldown(TRUE))
+                charge += NORMAL*traits[1].on_impact(src) 
+            if(manage_cooldown(TRUE))
+                visible_message("<span class='notice'>The [name] ticks.</span>")
+                check_charge()
+                if(prob(13))
+                    process_type = ""
+            charge = 0 //Don't really need to do this but, I am skeptical it may fix my bug. Coming back later, don't remeber if it did, too scared to change it now.
+        else    
+            return PROCESS_KILL
+
+/obj/item/xenoartifact/Destroy()
+    for(var/mob/living/C in contents) //mobs inside only have a 50/50 chance of surviving a collapse.
+        if(pick(FALSE, TRUE))
+            C.forceMove(get_turf(loc))
+        else
+            qdel(C)
+    ..()
+
+/obj/item/xenoartifact/maint //Semi-toddler-safe version for maint loot.
+    material = BLUESPACE
+
+/obj/item/xenoartifact/maint/Initialize(mapload, difficulty)
+    if(prob(0.1))
+        material = pick(PLASMA, URANIUM, BANANIUM)
+    difficulty = material
```

Comment:
What does setting the parameter difficulty here do?

Reply:
As a comment explains, it can be used by the listing console to set the material to a specific state. It's called difficulty because it represents how the console interacts with artifacts, pricing, & profit.